### PR TITLE
Added mobile/fill container mode 

### PR DESCRIPTION
**Pull request by katemenkhaus**
Introduces a mobile state for dt-picker. 
mobile mode = fillContainer = full screen

Changeds: 
- iron-media-query to determine when to force the picker to go into fillContainer mode. This was modeled after app-nav
- moved the time above the calendar in fullContainer mode to prevent ISO keyboard from covering the field
- `Polymer.AppLocalizeBehavior` is already include through the `PxDatetimeBehavior.TempMoment` 
- `prevent-cell-focus` - when fillContainer is true the first cell in the field should not be focused 
- removed used sass
- added styles for fillContainer mode

added properties: 
- `required` - added to match the other datetime components 
- `_fillContainer` - needed to separate the external `fillContainer` with the internal one to avoid over writing the external one when the screen becomes larger than the `collapseAt`
- `collapseAt` - the value at which the panel will be forced into the fillContainer mode. 
- `selectedDate` - used for the title when the panel is open in fillContainer mode
-`_belowCollapseSize` - set in `iron-media-query` will update `_fillContainer` as necessary

### CODE REVIEW

File: px-datetime-picker-content.html [hunk 1]
```diff
@@ -24,48 +23,55 @@
       position-target="[[_getPositionTarget(fillContainer, fitIntoElement)]]"
       fit-into="[[fitIntoElement]]">
       <div slot="dropdown-content" class="dt-container__box shadow-temporary">
-        <div class="dt-p-calendar">
+        <template is="dom-if" if="{{fillContainer}}">
```
Comment (nonmetalhail) [inline px-datetime-picker-content.html]: single bind

File: px-datetime-picker-content.html [hunk 2]
```diff
@@ -24,48 +23,55 @@
...
+          <div class="dt-p__header shadow-navigation">
+            <p class="dt-p__selected-date">{{selectedDate}}</p>
```
Comment (nonmetalhail) [inline px-datetime-picker-content.html]: single bind

File: px-datetime-picker-content.html [hunk 3]
```diff
@@ -24,48 +23,55 @@
...
+          </div>
+        </template>
+        <template id="timeTemplate" is="dom-if" if="{{_handleField(hideTime, fillContainer)}}">
```
Comment (nonmetalhail) [inline px-datetime-picker-content.html]: single bind

File: px-datetime-picker-content.html [hunk 4]
```diff
@@ -24,48 +23,55 @@
...
-                resources="[[resources]]"
-                language="[[language]]"
-                formats="[[formats]]"
-                min-date="[[minDate]]"
-                max-date="[[maxDate]]">
-              </px-datetime-field>
-            </div>
-          </template>
-          <template is="dom-if" if="{{showButtons}}">
+          <template is="dom-if" if="{{_handleShowButtons(showButtons, fillContainer)}}">
```
Comment (nonmetalhail) [inline px-datetime-picker-content.html]: single bind

File: sass/px-datetime-picker-content.scss [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+@import "px-helpers-design/_generic.helpers.scss";
+
+// Base
+
+//Objects
+@import 'px-actionable-design/_objects.actionable.scss';
+@import "px-shadows-design/_objects.shadows.scss";
+
+//Component
+// px-datetime-field {
```
Comment (nonmetalhail) [inline sass/px-datetime-picker-content.scss]: remove commented out sass

File: sass/px-datetime-picker-content.scss [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  position: absolute; //needs to be absolute to correctly position selected date
+}
+
+.dt-p__selected-date {
+  text-align: center;
+}
+
+/* Only exists in fill-container */
+.dt-p-time {
+  --px-datetime-entry-font-size: 30px;
```
Comment (nonmetalhail) [inline sass/px-datetime-picker-content.scss]: calcRem?

File: sass/px-datetime-picker-content.scss [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+:host([fill-container]) .dt-p-calendar {
+  margin-left: auto;
+  margin-right: auto;
+  display: block;
+  max-width: 500px;
+}
+
+// :host([fill-container]) .dt-p-time-buttons--wrapper{
```
Comment (nonmetalhail) [inline sass/px-datetime-picker-content.scss]: remove

File: sass/px-datetime-picker.scss [hunk 8]
```diff
@@ -43,70 +40,83 @@ px-datetime-field {
...
-  z-index: var(--px-datetime-picker-z-index, 300);
-  padding: $inuit-base-spacing-unit--small $inuit-base-spacing-unit--large $inuit-base-spacing-unit--large;
-}
-
-.dt-p-buttons {
-  display: flex;
-  justify-content: flex-end;
-  padding-top: $inuit-base-spacing-unit--small;
-}
+// .dt-p-today {
```
Comment (nonmetalhail) [inline sass/px-datetime-picker.scss]: remove

File: sass/px-datetime-picker.scss [hunk 9]
```diff
@@ -43,70 +40,83 @@ px-datetime-field {
...
-  --px-datetime-field-height: calculateRem(40px);
-  --px-btn-height: 1em;
-  --px-dropdown-content-font-size: 15px;
-  display: flex;
-  justify-content: center;
-  margin-top: $inuit-base-spacing-unit;
-  margin-bottom: $inuit-base-spacing-unit;
-}
 
+// :host([fill-container]) .dt-container__box {
```
Comment (nonmetalhail) [inline sass/px-datetime-picker.scss]: remove

**Comment by katemenkhaus** [comment]
- Consulted the designers about the time icon. It works as expected though we might take a look at it again soon. 
- Also discussed the fitIntoElement and collapseAt behavior with the designers. It works as expected. If the element is in a screen smaller than the collapseAt value then the content will override the fitIntoElement prop and go full screen.

**Pull request merged**